### PR TITLE
Header bug

### PR DESCRIPTION
**Pull request by AndrewCowle**
https://regulatorydelivery.atlassian.net/browse/COSBETA-1643

## Description
Moved the header link to partials for `search` and `submit` each landing on their own `homepages` and updated  `spec/requests/root_path_spec.rb` to ensure the correct partial is rendered for the given domain.

## Review apps

https://cosmetics-pr-2651-submit-web.london.cloudapps.digital/
https://cosmetics-pr-2651-search-web.london.cloudapps.digital/

## Checklist:

- [ ] Automated checks are passing locally.
- [ ] CHANGELOG updated if change is worth telling users about. Changes in notification wizard should always be included.
### General testing
- [ ] Test without javascript
- [ ] Test on small screen
### Accessibility testing
- [ ] Works keyboard only
- [ ] Tested with one screen reader
- [ ] Zoom page to 400% - content still visible
- [ ] Disable css - does content make sense and appear in a logical order?
- [ ] Passes automated checker (automated in build or manual)

### CODE REVIEW

**Review by nasirkhanbeis — APPROVED** [review]
Tested this on the review app, looking good as expected.

File: cosmetics-web/app/views/search/_header_link.html.erb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<a class="govuk-header__link govuk-header__link--service-name" href=<%= current_user.present? && current_user.has_completed_registration? ? poison_centre_notifications_search_path : "#" %>>
```
Comment (scruti) [inline cosmetics-web/app/views/search/_header_link.html.erb:1]:
```suggestion
<a class="govuk-header__link govuk-header__link--service-name" href=<%= current_user&.has_completed_registration? ? poison_centre_notifications_search_path : "#" %>>
```

File: cosmetics-web/app/views/submit/_header_link.html.erb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<a class="govuk-header__link govuk-header__link--service-name" href=<%= current_user.present? && current_user.has_completed_registration? ? products_redirect_responsible_persons_path : "#" %>>
```
Comment (scruti) [inline cosmetics-web/app/views/submit/_header_link.html.erb:1]: Suggestion: 

```suggestion
<a class="govuk-header__link govuk-header__link--service-name" href=<%= current_user&.has_completed_registration? ? products_redirect_responsible_persons_path : "#" %>>
```

File: cosmetics-web/spec/requests/root_path_spec.rb [hunk 3]
```diff
@@ -22,6 +22,11 @@
       it "renders the homepage" do
         expect(response).to render_template("submit/landing_page/index")
       end
+
+      it "renders the correct header link" do
+        get submit_root_path
+        expect(response).to render_template(partial: "submit/_header_link")
+      end
```
Comment (scruti) [inline cosmetics-web/spec/requests/root_path_spec.rb:29]: :clap:

**Review by scruti — APPROVED** [review]

**Pull request merged**